File: 0125-valid-palindrome/0125-valid-palindrome.cpp
#include <cctype>
#include <iostream>
#include <string>

class Solution {
public:
    bool isPalindrome(string s) {
        if (s.empty()) return true;

        string filtered; // To store valid characters

        // Process the string
        for (char c : s) {
            if (isalnum(c)) { // Check if character is alphanumeric
                filtered += tolower(c); // Convert to lowercase and add to filtered string
            }
        }

        // If filtered string is empty, return false
        if (filtered.empty()) return true;
        
        cout << filtered << endl;
        int L = 0;
        int R = filtered.length()-1;
        
        while(L < R){
            if(filtered[L] == filtered[R]){
                L++;
                R--;
            } else {
                return false;
            }
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        
        result = ""
        for char in s:
            char = char.lower()
            if char.isalnum():
                result += char
        
        left = 0
        right = len(result)-1
        while left < right:
            if result[right] != result[left]:
                return False
            right -= 1
            left += 1
        return True